ex5/s_tokida/main_mfcc.py

```diff
@@ -0,0 +1,184 @@
+import myfunc
+
+import argparse
+from tkinter import N
+import numpy as np
+import librosa
+import matplotlib.pyplot as plt
+from scipy.fftpack.realtransforms import dct
+
+def cal_m0(f0):
+    """calculate m0
+    Args:
+        f0 (float)
+    Returns:
+        m0 (float)
+    """
+    return 1000.0 / np.log(1000.0 / f0 + 1.0)
+
+def hztomel(f, f0=700.0):
+    """convert frequency [Hz] to mel frequency [mel]
+    Args:
+        f (ndarray): frequency [Hz]
+    Returns:
+        m (ndarray): mel frequency [mel]
+    """
+    m0 = cal_m0(f0)
+    return m0 * np.log(f / f0 + 1.0)
+
+def meltohz(m, f0=700.0):
+    """convert mel frequency [mel] to frequency [Hz]
+    Args:
+        m (ndarray): mel frequency [mel]
+    Returns:
+        f (ndarray): frequency [Hz]
+    """
+    m0 = cal_m0(f0)
+    return f0 * (np.exp(m / m0) -1.0)
+
+def melfilterbank(sr, shift_size, n_channel):
+    """make mel filter bank
+    Args:
+        sr (int): sample rate [Hz]
+        shift_size (int): shift size
+        n_channel (int): number of channel
+    Returns:
+        filterbank (ndarray): mel filter bank
+    """
+
+    fnyq = sr / 2  # nyquist frequency[Hz] = samplerate / 2
+    melnyq = hztomel(fnyq)  # nyquist frequency[mel]
+    nmax = shift_size // 2  # maximum number of frequency indexes
+    df = sr / shift_size  # frequency resolution (Hz width per frequency index)
+
+    dmel = melnyq / (n_channel + 1)  # center frequency of each filter on the Mel scale
+    melcenters = np.arange(1, n_channel + 1) * dmel
+    fcenters = meltohz(melcenters)
+    indexcenter = np.round(fcenters / df)  # Convert center frequency to frequency index
+    indexstart = np.hstack(([0], indexcenter[0:n_channel - 1]))
+    indexstop = np.hstack((indexcenter[1:n_channel], [nmax]))
+
+    filterbank = np.zeros((n_channel, nmax))
+
+    for c in range(0, n_channel):
+        # Find the point from the left of the triangular filter
+        increment= 1.0 / (indexcenter[c] - indexstart[c])
+        for i in range(int(indexstart[c]), int(indexcenter[c])):
+            filterbank[c, i] = (i - indexstart[c]) * increment
+        # Find the point from the right of the triangular filter
+        decrement = 1.0 / (indexstop[c] - indexcenter[c])
+        for i in range(int(indexcenter[c]), int(indexstop[c])):
+            filterbank[c, i] = 1.0 - ((i - indexcenter[c]) * decrement)
```

Comment:
numpy.linspace を使えば簡単に書けます（index~は全部int型とします）
```suggestion
        filterbank[c, indexstart[c]:indexcentor[c]] = np.linspace(0, 1, indexcentor[c] - indexstart[c], endpoint=False)
        filterbank[c, indexcentor[c]:indexstop[c]] = np.linspace(1, 0, indexstop[c] - indexcentor[c], endpoint=False)
```